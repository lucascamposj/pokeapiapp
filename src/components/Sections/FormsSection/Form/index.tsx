import React from 'react';
import { Container, FormText } from './styles';
import { FormInterface } from '../../../../pages/Pokemon';

const Form: React.FC<FormInterface> = ({ name }: FormInterface) => {
  return (
    <Container>
      <FormText>{name}</FormText>
    </Container>
  );
};

export default Form;
