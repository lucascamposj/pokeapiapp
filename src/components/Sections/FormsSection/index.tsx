import React from 'react';
import Form from './Form';
import { Container } from './styles';
import { FormInterface } from '../../../pages/Pokemon';

interface FormSectionProps {
  forms: FormInterface[] | undefined;
}

const FormSection: React.FC<FormSectionProps> = ({
  forms,
}: FormSectionProps) => {
  return (
    <Container>
      {forms?.map(({ name, url }) => (
        <Form key={name} name={name} url={url} />
      ))}
    </Container>
  );
};

export default FormSection;
