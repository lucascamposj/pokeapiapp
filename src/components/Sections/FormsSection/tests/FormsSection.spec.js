import React from 'react';
import {shallow} from 'enzyme';
import FormsSection from '../../FormsSection';

describe('FormsSection', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const forms = [
              {
                name: "Overgrow",
                url: "url-test.com"
              },
              {
                name: "Chlorophyll",
                url: "url-test.com"
              }
            ]
            const component = shallow(
              <FormsSection
                forms={forms}
              />)
            expect(component).toMatchSnapshot()
        });
    });
});
