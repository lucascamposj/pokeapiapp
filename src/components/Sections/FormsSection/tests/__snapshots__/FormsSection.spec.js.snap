// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormsSection Rendering should match to snapshot 1`] = `
<Styled(View)>
  <Form
    key="Overgrow"
    name="Overgrow"
    url="url-test.com"
  />
  <Form
    key="Chlorophyll"
    name="Chlorophyll"
    url="url-test.com"
  />
</Styled(View)>
`;
